src/File.php

```diff
@@ -46,6 +46,9 @@ class File
     /** @var int */
     protected $fileSize;
 
+    /** @var string[] */
+    protected $metaInfo = [];
```

Comment:
Yeah, I was pretty much following the convention.

Honestly, keeping all those fields as `protected` and - by extension - considering a public API (since any change to them would break anyone extending those classes and using those fields) prevents adjusting internal structure. They should be dropped in favor of accessors.

If repository maintainer @ankitpokhrel would accept it, I'd say any new properties introduced should be private and all fields being protected should be considered deprecated in terms of their scope.

Reply:
Switched to `private`.